### PR TITLE
Fix for memory leak while parsing AuthorityKeyId ext

### PR DESCRIPTION
**Pull request by mprse**
## Description

Resolves: https://github.com/Mbed-TLS/mbedtls/issues/7576

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [X] **changelog** not required (bug is in unreleased code)
- [X] **backport** not required (bug affects code not in `mbedtls-2.28`)
- [X] **tests** provided

### CODE REVIEW

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

Note that we may want to replace these testcases with constructed ones in a followup as happened in #6949.

**Review by davidhorstmann-arm — DISMISSED** [review]
LGTM

**Comment by AndrzejKurek** [comment]
There are some CI issues - I'll wait with review until there are no issues.

File: tests/suites/test_suite_x509parse.data [hunk 1]
```diff
@@ -3314,3 +3314,8 @@ x509_crt_parse_authoritykeyid:"308203873082026fa003020102020100300d06092a864886f
...
+
+# clusterfuzz-testcase-minimized-fuzz_x509crt-6666050834661376: test for bad sequence of names in authorityCertIssuer
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_x509parse.data:3318]: It would be great to state in more detail what's wrong.

**Review by AndrzejKurek — APPROVED** [review]
Only a minor issue found, not worth pushing on its own.

File: tests/suites/test_suite_x509parse.data (hunk 1 above)
Comment (mprse) [reply tests/suites/test_suite_x509parse.data:3318]: I was thinking how much info to add here and I think that details can be checked in the issue when someone is interested in changing this test case.

Comment (AndrzejKurek) [reply tests/suites/test_suite_x509parse.data:3318]: But the issue is not mentioned here :)

Comment (mprse) [reply tests/suites/test_suite_x509parse.data:3318]: Yes, but it is easy to find PR that added this line and it links directly to the issue.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM

**Comment by gilles-peskine-arm** [comment]
Travis runs job that don't run on Jenkins, so if those error out, there's a risk. I restarted the erroring jobs, we should have updated results in a few minutes.

**Comment by mprse** [comment]
>Travis runs job that don't run on Jenkins, so if those error out, there's a risk. I restarted the erroring jobs, we should have updated results in a few minutes.

All green!

**Pull request merged**